### DISCUSSION TITLE
Should avoid the use of getInitialProps in _app.js for hybrid?

### DISCUSSION THREAD

**Discussion by matamatanot** [vercel/next.js#10704] [Help] [answered]
I have read #9524 and understood static and  hybrid advantage.

To use Redux, I use next-redux-wrapper now.
[kirill-konshin/next-redux-wrapper: Redux wrapper for Next.js](https://github.com/kirill-konshin/next-redux-wrapper)
This library is not compatible with Auto Partial Static Export feature because of getInitialProps in _app.js
In the future, I want to use getStaticProps and getStaticPaths to build a hybrid application (JAMstack?).

Is it better not to use getInitialProps in _app.js?
Is Using Redux incompatible with hybrid applications?

I'm excited about SSG Improvements and new APIs. 👍

**Comment by timneutkens** [comment, marked as answer]
The examples will be updated, in general you don't need redux though.

**Reply by guigrpa** [reply]
In our website, several of the pages share a Redux store. Others are static and do not need Redux. In this hybrid site, we were using next-redux-wrapper and it worked fine. Now we want to migrate to `getStaticPaths()` and `getStaticProps()`, but they are apparently incompatible with next-redux-wrapper, maybe due to what @matamatanot posted above.

We've also explored other options in your examples (currently using `getInitialProps()`), modifying them to use `getStaticProps()`, but to no avail. Maybe you can point to a working example with Redux and `getStaticProps()`? Thanks

**Reply by samao** [reply]
getServerSideProps and getStaticProps are executed earlier than store creation, so there is no way to get the store inside these two methods. These new APIs don't seem to be suitable for redux
